Gems/Prefab/PrefabDependencyViewer/Code/Source/PrefabDependencyViewerEditorSystemComponent.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzToolsFramework/API/ViewPaneOptions.h>
+#include <PrefabDependencyViewerEditorSystemComponent.h>
+#include <MainWindow.h>
+#include <QMenu>
+#include <LyViewPaneNames.h>
+#include <PrefabDependencyTree.h>
+
+namespace PrefabDependencyViewer
+{
+    using Outcome = AZ::Outcome<PrefabDependencyTree, const char*>;
+
+    InstanceEntityMapperInterface* PrefabDependencyViewerEditorSystemComponent::s_prefabEntityMapperInterface = nullptr;
+    PrefabSystemComponentInterface* PrefabDependencyViewerEditorSystemComponent::s_prefabSystemComponentInterface = nullptr;
+    PrefabPublicInterface* PrefabDependencyViewerEditorSystemComponent::s_prefabPublicInterface = nullptr;
+
+    const char* PrefabDependencyViewerEditorSystemComponent::s_prefabViewerTitle = "Prefab Dependencies Viewer";
+
+    void PrefabDependencyViewerEditorSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<PrefabDependencyViewerEditorSystemComponent, AZ::Component>()
+                ->Version(0);
+        }
+    }
+
+    PrefabDependencyViewerEditorSystemComponent::PrefabDependencyViewerEditorSystemComponent() = default;
+
+    void PrefabDependencyViewerEditorSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("PrefabDependencyViewerEditorService"));
+    }
+
+    void PrefabDependencyViewerEditorSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("PrefabDependencyViewerEditorService"));
+    }
+
+    void PrefabDependencyViewerEditorSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+    }
+
+    void PrefabDependencyViewerEditorSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
+
+    void PrefabDependencyViewerEditorSystemComponent::Activate()
+    {
+        s_prefabEntityMapperInterface = AZ::Interface<InstanceEntityMapperInterface>::Get();
```

Comment:
If you depend on this interface existing you should declare that you depend on "EditorEntityContextService" in your GetRequiredServices.  Looking at the code, the EditorEntityContextComponent creates the class that services this interface, and EditorEntityContextComponent publishes a service called "EditorEntityContextService"

Reply:
Note that this dll, especially if its listed as "Tools" might be loaded into tools that are not the editor, in which case instead of asserting it may be best to gracefully return.

Reply:
Resolved.

Reply:
One question I have though is that I have listed the EditorEntityContextService as a required service => I'd still get an error and the component wouldn't get activated if it's loaded outside the Editor. Is that ok?

Reply:
As a Required service means that if the thing it depends on is missing the entire application won't start.  A box will pop up and the application will shut down.